modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/service/test/ObjectFieldLocalServiceTest.java

```diff
@@ -1094,6 +1095,106 @@ public void testUpdateCustomObjectField() throws Exception {
 			objectDefinition.getObjectDefinitionId());
 	}
 
+	@Test
+	public void testUpdateRequiredRelationshipObjectField() throws Exception {
```

Comment:
Your test looks great, but I think we should do the following in the given order:
Add two definitions;
Add one cascade relationship;
Change the required of the field to true;
Assert;
Change the relationship to disassociate;
Assert that the field changed to required false;
Try to change it to true and assert that you couldn't;
Change the relationship to prevent;
Change the field required to true;
Assert;

You don't need to publish definitions here. According to your other PRs, activation status of a definition doesn't matter for that.